### PR TITLE
Add billing process and chain bus support

### PR DESCRIPTION
**Pull request by zeqing-guo**
Add billing process and chain bus support.

### CODE REVIEW

File: blockproducer/chain.go [hunk 1]
```diff
@@ -677,6 +697,14 @@ func (c *Chain) loadAccountStableBalance(addr proto.AccountAddress) (balance uin
 	return c.immutable.loadAccountStableBalance(addr)
 }
 
+func (c *Chain) loadSQLChainProfile(databaseID proto.DatabaseID) (profile *types.SQLChainProfile, ok bool) {
+	c.RLock()
+	defer c.RUnlock()
+	profileObj, ok := c.immutable.loadSQLChainObject(databaseID)
+	profile = &profileObj.SQLChainProfile
```
Comment (leventeliu) [inline blockproducer/chain.go]: Need to copy this object, it's not immutable now.

File: worker/chainbusservice_test.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		)
+		privKey, err = kms.GetLocalPrivateKey()
+		So(err, ShouldBeNil)
+		pubKey = privKey.PubKey()
+		addr, err = crypto.PubKeyHash(pubKey)
+		So(err, ShouldBeNil)
+		ctx, _ := context.WithCancel(context.Background())
+		bs := NewBusService(ctx, addr, testCheckInterval)
+		topic := fmt.Sprintf("/%s/", testOddBlocks.Transactions[0].GetTransactionType().String())
+		var count uint32 = 0
```
Comment (covenantdb) [inline worker/chainbusservice_test.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
should drop = 0 from declaration of var count; it is the zero value

**Review by auxten — APPROVED** [review]
so long, Approve

**Review by xq262144 — APPROVED** [review]
too long, approve

**Pull request merged**